# Exploritory data sift

In [2]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

## ArcGIS Combined Spatial Data

For the sake of time, I combined both Tract and County census data using ArcGIS's spatial join geoprocessing tool.
The join was one to many and the resulting table was exported into an xls format.

In [25]:
df = pd.read_excel('data/demographic_spatial_join.xls')
df.ZCTA5CE10.value_counts()

98001    1
98122    1
98155    1
98154    1
98148    1
        ..
98050    1
98047    1
98045    1
98043    1
98940    1
Name: ZCTA5CE10, Length: 101, dtype: int64

In [26]:
# Zip codes in King County that are included in housing data

kc_zips = [98178, 98125, 98028, 98136, 98074, 98053, 98003, 98198,
           98146, 98038, 98007, 98115, 98107, 98126, 98019, 98103, 
           98002, 98133, 98040, 98092, 98030, 98119, 98112, 98052, 
           98027, 98117, 98058, 98001, 98056, 98166, 98023, 98070, 
           98148, 98105, 98042, 98008, 98059, 98122, 98144, 98004, 
           98005, 98034, 98075, 98116, 98010, 98118, 98199, 98032, 
           98045, 98102, 98077, 98108, 98168, 98177, 98065, 98029, 
           98006, 98109, 98022, 98033, 98155, 98024, 98011, 98031, 
           98106, 98072, 98188, 98014, 98055, 98039]

In [27]:
# Filter demographic data for zip codes from housing data

b_series = df.ZCTA5CE10.isin(kc_zips)
df_zips = df[b_series]

df_zips.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 70 entries, 0 to 85
Columns: 131 entries, OBJECTID to Shape_Area
dtypes: float64(37), int64(91), object(3)
memory usage: 72.2+ KB


In [29]:
df_names = df_zips[['ZCTA5CE10', 'NAME']]

<class 'pandas.core.frame.DataFrame'>
Int64Index: 70 entries, 0 to 85
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   ZCTA5CE10  70 non-null     int64 
 1   NAME       70 non-null     object
dtypes: int64(1), object(1)
memory usage: 1.6+ KB


## WA Tract Census Data

In [2]:
df = pd.read_csv('data/WSDOT_ALPACA_Census_Tract_2015.csv')

In [3]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2570 entries, 0 to 2569
Data columns (total 272 columns):
 #    Column                          Dtype  
---   ------                          -----  
 0    OBJECTID                        int64  
 1    STATE                           int64  
 2    COUNTY                          int64  
 3    TRACT                           int64  
 4    BLKGRP                          float64
 5    GEOID                           int64  
 6    ALAND                           int64  
 7    AWATER                          int64  
 8    ALAND_SQMI                      float64
 9    Table_ID_RACE                   object 
 10   Title_RACE                      object 
 11   Universe_RACE                   object 
 12   Total_RACE                      int64  
 13   OneRace                         int64  
 14   White                           int64  
 15   NotWhite                        int64  
 16   PctNotWhite                     float64
 17   AfricanAmeri

Lots of columns (272).

We will look at specific data based on column descriptions from WSDOT_ALPACA_Census_Tract_2015.pdf

We are interested in columns for:

- Income - Estimated Median Income at the Household level
- F_22to24 - Total females ages 22 to 24
- F_25to29 - Total females ages 25 to 29
- F_30to34 - Total females ages 30 to 34
- F_35to39 - Total females ages 35 to 39
- F_40to44 - Total females ages 40 to 44
- F_45to49 - Total females ages 45 to 49
- F_50to54 - Total females ages 50 to 54
- F_55to59 - Total females ages 55 to 59
- M_22to24 - Total males ages 22 to 24
- M_25to29 - Total males ages 25 to 29
- M_30to34 - Total males ages 30 to 34
- M_35to39 - Total males ages 35 to 39
- M_40to44 - Total males ages 40 to 44
- M_45to49 - Total males ages 45 to 49
- M_50to54 - Total males ages 50 to 54
- M_55to59 - Total males ages 55 to 59
- Total_RACE - Total estimated population for race table
- Total_AGE - Total estimated population for age table
- POPDENSITY - A calculated field derived from dividing Total_AGE by the ALAND_SQMI field.
- COUNTY - King County FIPS code is 033

In [4]:
# Defining a list of columns we are interested in.
Col_Int = ['Income', 'COUNTY', 'GEOID', 
           'F_22to24', 'F_25to29', 'F_30to34', 'F_35to39', 'F_40to44', 'F_45to49', 'F_50to54', 'F_55to59',
           'M_22to24', 'M_25to29', 'M_30to34', 'M_35to39', 'M_40to44', 'M_45to49', 'M_50to54', 'M_55to59',
           'Total_AGE', 'POPDENSITY']

In [15]:
# Check for naming error
def check_list(df,cols):
    for i in cols:
        if i not in df.columns:
            print(f'{i} not in columns')
    print('Process Complete')
    return

In [32]:
check_list(df,Col_Int)

Process Complete


In [7]:
# Selecting columns from list of interest
df_clean = df[Col_Int]

In [8]:
# Filtering for King County (FIPS 033)
df_kc_clean = df_clean[df_clean['COUNTY'] == 33]
df_kc_clean.head()

Income  COUNTY        GEOID  F_22to24  F_25to29  F_30to34  F_35to39  \
966  163594      33  53033024200        42        14        13        96   
967  108406      33  53033022203       147       208       151       156   
968   70180      33  53033024800       116       223       276       209   
969  103449      33  53033024902       169        41        38       195   
970  149984      33  53033024903       126        65       104       262   

     F_40to44  F_45to49  F_50to54  ...  M_22to24  M_25to29  M_30to34  \
966       161       127        74  ...        33        30         0   
967       301       252       196  ...        16       214       137   
968       251       291       190  ...        71       272       177   
969       189       145       169  ...        25       111        21   
970       304       355       366  ...        18        94       112   

     M_35to39  M_40to44  M_45to49  M_50to54  M_55to59  Total_AGE  POPDENSITY  
966        77        60       156       119        70       2906    1.255569  
967       151       304       273       240       215       5135    2.578054  
968       221       257       304       280       171       5915    2.521194  
969       125       243       159       184       118       4200    3.329703  
970       114       297       252       327       440       6923    2.244649  

[5 rows x 21 columns]

In [27]:
df_kc_clean['GEOID10'] = df_kc_clean['GEOID']/10000

df_kc_clean.style.set_precision(0)

df_kc_clean['GEOID10'] = df_kc_clean['GEOID10'].astype(int)

df_kc_clean

C:\Users\bjere\AppData\Local\Temp/ipykernel_22712/3527643848.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_kc_clean['GEOID10'] = df_kc_clean['GEOID']/10000
C:\Users\bjere\AppData\Local\Temp/ipykernel_22712/3527643848.py:3: FutureWarning: this method is deprecated in favour of `Styler.format(precision=..)`
  df_kc_clean.style.set_precision(0)
C:\Users\bjere\AppData\Local\Temp/ipykernel_22712/3527643848.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_kc_clean['GEOID10'] = df_kc_clean['GEOID10'

Income  COUNTY        GEOID  F_22to24  F_25to29  F_30to34  F_35to39  \
966   163594      33  53033024200        42        14        13        96   
967   108406      33  53033022203       147       208       151       156   
968    70180      33  53033024800       116       223       276       209   
969   103449      33  53033024902       169        41        38       195   
970   149984      33  53033024903       126        65       104       262   
...      ...     ...          ...       ...       ...       ...       ...   
1867   46106      33  41033360100        17        11        55        34   
1868   25008      33  41033361600        60       263       209       175   
1869   50198      33  41033360400         0        87        16        62   
1873   39930      33  41033361400        29        27        76        32   
2531   35833      33  16033950100        26        21        29        27   

      F_40to44  F_45to49  F_50to54  ...  M_25to29  M_30to34  M_35to39  \
966        161       127        74  ...        30         0        77   
967        301       252       196  ...       214       137       151   
968        251       291       190  ...       272       177       221   
969        189       145       169  ...       111        21       125   
970        304       355       366  ...        94       112       114   
...        ...       ...       ...  ...       ...       ...       ...   
1867        90        34       148  ...        43        69        66   
1868       185       278       241  ...       158       294       111   
1869       133       124       195  ...        34        98        60   
1873        80       105        80  ...        38        46       112   
2531         8        46        21  ...        27         6        34   

      M_40to44  M_45to49  M_50to54  M_55to59  Total_AGE  POPDENSITY  GEOID10  
966         60       156       119        70       2906    1.255569  5303302  
967        304       273       240       215       5135    2.578054  5303302  
968        257       304       280       171       5915    2.521194  5303302  
969        243       159       184       118       4200    3.329703  5303302  
970        297       252       327       440       6923    2.244649  5303302  
...        ...       ...       ...       ...        ...         ...      ...  
1867        41       130       106       102       2682    0.004775  4103336  
1868       150       157       367       281       7351    0.010018  4103336  
1869       115       111       273       149       3742    0.027781  4103336  
1873        42        81        43       114       2688    0.021509  4103336  
2531         0        22        33        43        780    0.000275  1603395  

[414 rows x 22 columns]

In [28]:
df_kc_clean['GEOID10'].value_counts()

5303302    162
5303300    104
5303303    103
5303301     28
4103336     16
1603395      1
Name: GEOID10, dtype: int64

## WA Census Data By ZIP

In [10]:
df2 = pd.read_csv('data/LIHEAPMap.csv')

In [11]:
df2.columns.tolist()

['OBJECTID_1',
 'OBJECTID',
 'ZCTA5CE10',
 'GEOID10',
 'MTFCC10',
 'FUNCSTAT10',
 'PARTFLG10',
 'INTPTLON10',
 'INTPTLAT10',
 'Version',
 'POP2000',
 'POP2001',
 'POP2002',
 'POP2003',
 'POP2004',
 'POP2005',
 'POP2006',
 'POP2007',
 'POP2008',
 'POP2009',
 'POP2010',
 'POP2011',
 'POP2012',
 'POP2013',
 'POP2014',
 'POP2015',
 'POP2016',
 'POP2017',
 'POP2018',
 'POP2019',
 'HHP2000',
 'HHP2001',
 'HHP2002',
 'HHP2003',
 'HHP2004',
 'HHP2005',
 'HHP2006',
 'HHP2007',
 'HHP2008',
 'HHP2009',
 'HHP2010',
 'HHP2011',
 'HHP2012',
 'HHP2013',
 'HHP2014',
 'HHP2015',
 'HHP2016',
 'HHP2017',
 'HHP2018',
 'HHP2019',
 'GQ2000',
 'GQ2001',
 'GQ2002',
 'GQ2003',
 'GQ2004',
 'GQ2005',
 'GQ2006',
 'GQ2007',
 'GQ2008',
 'GQ2009',
 'GQ2010',
 'GQ2011',
 'GQ2012',
 'GQ2013',
 'GQ2014',
 'GQ2015',
 'GQ2016',
 'GQ2017',
 'GQ2018',
 'GQ2019',
 'HU2000',
 'HU2001',
 'HU2002',
 'HU2003',
 'HU2004',
 'HU2005',
 'HU2006',
 'HU2007',
 'HU2008',
 'HU2009',
 'HU2010',
 'HU2011',
 'HU2012',
 'HU2013',
 'HU2014'

In [12]:
df2.head()

OBJECTID_1  OBJECTID  ZCTA5CE10  GEOID10 MTFCC10 FUNCSTAT10 PARTFLG10  \
0           1         1      98001  5398001   G6350          S         N   
1           2         2      98002  5398002   G6350          S         N   
2           3         3      98003  5398003   G6350          S         N   
3           4         4      98004  5398004   G6350          S         N   
4           5         5      98005  5398005   G6350          S         N   

   INTPTLON10  INTPTLAT10             Version  ...  sum_housetypemobile  \
0 -122.264469   47.309496  September 10, 2019  ...                   40   
1 -122.216812   47.308286  September 10, 2019  ...                  284   
2 -122.315773   47.307713  September 10, 2019  ...                  282   
3 -122.205341   47.618337  September 10, 2019  ...                    0   
4 -122.168798   47.614533  September 10, 2019  ...                    0   

   sum_housetyperv  sum_ownedorbuying  sum_subsidized  sum_rental  \
0                0                266             815         267   
1                3                403            1587        1187   
2                0                512            3215        1864   
3                0                 75             586         504   
4                0                103             272         237   

   sum_roomerboard  sum_temporaryhousing  Point_Count  Shape__Area  \
0                0                     0         1348     0.005825   
1                0                     0         3177     0.002244   
2                0                     0         5591     0.003613   
3                0                     0         1165     0.002678   
4                4                     0          616     0.002327   

   Shape__Length  
0       0.432608  
1       0.330129  
2       0.389413  
3       0.286314  
4       0.307980  

[5 rows x 160 columns]

Data needs cleaning. We're only interested in zip coded areas for King County (70 zip codes)

These zip codes will be merged later into bins.

Need to create a GEOID10 column for previous list with just the first 10 digits of the GEOID column with which to join
the two data sets.

Need to seperate out colums of interest for this data set.

Columns of interest are:

- GEOID10: For joining other data set
- ZCTA5CE10: 5-digit zip codes
- POP2000 - POP2019: population sizes by zipcode for years between 2000 and 2019


In [35]:
Col_Int_LIHEAP = ['GEOID10', 'ZCTA5CE10',
                  'POP2000', 'POP2001', 'POP2002', 'POP2003', 'POP2004', 'POP2005', 'POP2006', 'POP2007',
                  'POP2008', 'POP2009', 'POP2010', 'POP2011', 'POP2012', 'POP2013', 'POP2014', 'POP2015',
                  'POP2016', 'POP2017', 'POP2018', 'POP2019']

In [36]:
check_list(df2, Col_Int_LIHEAP)

Process Complete


In [38]:
# New dataframe with just columns of interest.
df2_int = df2[Col_Int_LIHEAP]

df2_int.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 598 entries, 0 to 597
Data columns (total 22 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   GEOID10    598 non-null    int64  
 1   ZCTA5CE10  598 non-null    int64  
 2   POP2000    598 non-null    float64
 3   POP2001    598 non-null    float64
 4   POP2002    598 non-null    float64
 5   POP2003    598 non-null    float64
 6   POP2004    598 non-null    float64
 7   POP2005    598 non-null    float64
 8   POP2006    598 non-null    float64
 9   POP2007    598 non-null    float64
 10  POP2008    598 non-null    float64
 11  POP2009    598 non-null    float64
 12  POP2010    598 non-null    int64  
 13  POP2011    598 non-null    float64
 14  POP2012    598 non-null    float64
 15  POP2013    598 non-null    float64
 16  POP2014    598 non-null    float64
 17  POP2015    598 non-null    float64
 18  POP2016    598 non-null    float64
 19  POP2017    598 non-null    float64
 20  POP2018   

In [40]:
# Filtering by zipcodes just in King County

b_series = df2_int.ZCTA5CE10.isin(kc_zips)
df2_int_filtered = df2_int[b_series]

df2_int_filtered.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 70 entries, 0 to 92
Data columns (total 22 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   GEOID10    70 non-null     int64  
 1   ZCTA5CE10  70 non-null     int64  
 2   POP2000    70 non-null     float64
 3   POP2001    70 non-null     float64
 4   POP2002    70 non-null     float64
 5   POP2003    70 non-null     float64
 6   POP2004    70 non-null     float64
 7   POP2005    70 non-null     float64
 8   POP2006    70 non-null     float64
 9   POP2007    70 non-null     float64
 10  POP2008    70 non-null     float64
 11  POP2009    70 non-null     float64
 12  POP2010    70 non-null     int64  
 13  POP2011    70 non-null     float64
 14  POP2012    70 non-null     float64
 15  POP2013    70 non-null     float64
 16  POP2014    70 non-null     float64
 17  POP2015    70 non-null     float64
 18  POP2016    70 non-null     float64
 19  POP2017    70 non-null     float64
 20  POP2018    7